test/basic_object_test.rb

```diff
@@ -0,0 +1,25 @@
+require 'test_helper'
+require 'hanami/utils/basic_object'
+
+class TestClass < Hanami::Utils::BasicObject
+end
+
+describe Hanami::Utils::BasicObject do
+  describe '#respond_to_missing?' do
+    it 'raises an exception if respond_to? method is not implemented' do
+      -> {
+        TestClass.new.respond_to?(:no_existing_method)
+      }.must_raise(NotImplementedError)
+    end
+
+    it 'must return true if respond_to? method was implemented' do
```

Comment:
@joneslee85, so your suggestion is: `returns true given respond_to? method was implemented`?